packages/aws-cdk/lib/context-providers/index.ts

```diff
@@ -28,7 +29,14 @@ export async function provideContextValues(
 
     const provider = new constructor(sdk);
 
-    const value = await provider.getValue(missingContext.props);
+    let value;
+    try {
+      value = await provider.getValue(missingContext.props);
```

Comment:
I feel like I am missing something here. Why not just set the key to `undefined` and let the system stabilize itself?

Reply:
First, because we couldn't tell the difference between a value that hasn't been looked up yet, and a value that we tried to look up but the lookup failed.

Second, because the point is to catch the error and transport it to the CX.

Reply:
Not sure I understand the response. Why do we need to tell the difference?

Reply:
At some point I used to tell the difference between failed lookups and untried lookups, that's no longer the case. But we still need to transport the error to the CX, so it can attach a useful error message to the construct tree.